### ISSUE TITLE
is there a tool  can split custom obb data to train and val，evenly divided by classes

### ISSUE THREAD

**Issue by jaffe-fly** [ultralytics/ultralytics#18238]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

is there a tool  can split custom obb data to train and val，evenly divided by classes, not random split?

### Additional

no

**Comment by UltralyticsAssistant** [comment]
👋 Hello @jaffe-fly, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, as well as explanations for many commonly asked questions.

If this is a ❓ Question, please provide as much context as possible to help us assist you better. In your case, since you're asking about splitting custom OBB data evenly by classes for train and val, could you please elaborate further? For example:
- What kind of data format are you using?
- Are you following any specific annotation schema (e.g., COCO, YOLO, etc.)?
- Do you already have a splitting method in place that you'd like us to modify or enhance?

If this is a 🐛 Bug Report, and you're experiencing issues with implementation, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it.

For now, here are a few steps you might consider to proceed:
1. Upgrade to the latest `ultralytics` version to ensure full compatibility and access to the newest features:
```bash
pip install -U ultralytics
```
2. Explore our official guides and tips to confirm you are adhering to our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best! 🎉 For real-time conversations, head over to [Discord](https://discord.com/invite/ultralytics) 🎧. For more detailed discussions, check out [Discourse](https://community.ultralytics.com). Or share and explore ideas with others on [Reddit](https://reddit.com/r/Ultralytics).

## Tools and Environments

YOLO supports various verified environments (to ensure a smooth process, pre-installed dependencies like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) may be required):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

An Ultralytics engineer will review your request and provide further assistance soon. Thank you for your patience and for being a part of the Ultralytics community! 🚀

**Comment by glenn-jocher** [comment]
@jaffe-fly ultralytics does not currently provide a built-in tool specifically for class-balanced splits of custom OBB data. However, you can use libraries like `scikit-learn`'s `StratifiedShuffleSplit` to achieve this. Alternatively, you can explore the `autosplit` function in our repository, but note it performs random splits, not class-balanced ones. For details, refer to [autosplit documentation](https://docs.ultralytics.com/reference/data/utils/#autosplit).

**Comment by jaffe-fly** [comment]
used curosr to gen a script, anyone can try

**Comment by glenn-jocher** [comment]
For class-balanced splits of custom OBB data, we recommend implementing stratified splitting using `scikit-learn` as shown in our [K-Fold Cross Validation Guide](https://docs.ultralytics.com/guides/kfold-cross-validation/). While Ultralytics doesn't provide a built-in class-balanced split tool, this method ensures even class distribution across splits.
